starter_code/src/movies.js

```diff
@@ -3,18 +3,89 @@
 
 
 // Get the average of all rates with 2 decimals 
+function ratesAverage(movies){
+    var totalrate = movies.reduce(function(sum,oneMovie){
+        return sum + parseFloat(oneMovie.rate);
+    
+    },0);
+    var averagerate = totalrate/movies.length;
+    console.log(averagerate);
+    return averagerate
+}
+
+ratesAverage(movies);
+
+
 
 
 // Get the average of Drama Movies
+  
+function dramaMoviesRate(movies){
+    
+
+    var dramamovies = movies.filter(function(movie){
+       return movie.genre.indexOf("Drama")>= 0
+        
+    })
+    console.log(dramamovies);  
+  
+  
+        var dramaMoviesTotalRate = dramamovies.reduce(function(sum,onedramaMovie){
+          return sum + Number(onedramaMovie.rate);
+      },0)
+      console.log(dramaMoviesTotalRate);
+  
+      var dramaMovieAveragerate = parseFloat(dramaMoviesTotalRate)/dramamovies.length
+      console.log(dramaMovieAveragerate);
+      return parseFloat(dramaMovieAveragerate);
+  
+  }
```

Comment:
To get rid of the to errors you still have on Jasmine, you just have to add '.toFixed(2)' to the final result, to trunk it to 2 decimals, and to add and if statement to handle the case where there are no drama movies in the array : 
```

function dramaMoviesRate(movies) {
  var dramamovies = movies.filter(function(movie) {
    return movie.genre.indexOf("Drama") >= 0;
  });
  console.log(dramamovies);

  if (dramamovies.length === 0) {
    return undefined;
  }

  var dramaMoviesTotalRate = dramamovies.reduce(function(sum, onedramaMovie) {
    return sum + Number(onedramaMovie.rate);
  }, 0);
  console.log(dramaMoviesTotalRate);

  var dramaMovieAveragerate =
    parseFloat(dramaMoviesTotalRate) / dramamovies.length;
  console.log(dramaMovieAveragerate);
  return parseFloat(dramaMovieAveragerate.toFixed(2));
}
```